src/wp-includes/class-wp-theme-json-resolver.php

```diff
@@ -582,8 +593,8 @@ public static function get_user_global_styles_post_id() {
 	public static function theme_has_support() {
 		if ( ! isset( static::$theme_has_support ) ) {
 			static::$theme_has_support = (
-				is_readable( static::get_file_path_from_theme( 'theme.json' ) ) ||
-				is_readable( static::get_file_path_from_theme( 'theme.json', true ) )
+				static::get_file_path_from_theme( 'theme.json' ) !== '' ||
+				static::get_file_path_from_theme( 'theme.json', true ) !== ''
```

Comment:
I don't think this is needed here. Yoda conditions are only required around variable to value comparisons, not with functions (since setting a function to another value would result in an error anyway).